src/cogs/mirror/mirror.py

```diff
@@ -6,33 +6,35 @@
 
 class Mirror(commands.Cog):
 
-    def __init__(self, bot):
-        self.bot = bot
+    def __init__(self, bot, cache: dict):
+        self.bot: commands.Bot = bot
+        self.cache: dict = cache
 
     @commands.Cog.listener()
     async def on_message(self, message: discord.Message):
-        if 'mirror_cache' in self.bot.cache.keys() and message.channel.id in self.bot.cache['mirror_cache'].keys():
-            mirror_channel_id = self.bot.cache['mirror_cache'][message.channel.id]
+        if 'mirror_cache' in self.cache.keys() and message.channel.id in self.cache['mirror_cache'].keys():
+            mirror_channel_id = self.cache['mirror_cache'][message.channel.id]
             await message.guild.get_channel(mirror_channel_id).send(
                 f"[{message.created_at}] [{message.channel.name}] \n[{message.author.name}] : {message.content}")
 
     # This is mainly for testing
     @commands.command(name='cache')
     async def cache(self, ctx: Context, *args):
         if len(args) == 0:
-            await ctx.channel.send(self.bot.cache)
+            await ctx.channel.send(self.cache)
         elif len(args) == 1:
-            if args[0] in self.bot.cache.keys():
-                await ctx.channel.send(self.bot.cache[args[0]])
+            if args[0] in self.cache.keys():
+                await ctx.channel.send(self.cache[args[0]])
         else:
-            self.bot.cache[args[0]] = ' '.join(str(arg) for arg in args[1:])
+            self.cache[args[0]] = ' '.join(str(arg) for arg in args[1:])
 
     @commands.command(name='mirror')
     async def mirror(self, ctx: Context):
-        if 'mirror_cache' not in self.bot.cache.keys():
-            self.bot.cache['mirror_cache']: dict = {ctx.message.channel.id: ctx.message.channel_mentions[0].id}
+        if 'mirror_cache' not in self.cache.keys():
+            self.cache['mirror_cache']: dict = {ctx.message.channel.id: ctx.message.channel_mentions[0].id}
         else:
-            self.bot.cache['mirror_cache'][ctx.message.channel.id] = ctx.message.channel_mentions[0].id
+            self.cache['mirror_cache'][ctx.message.channel.id] = ctx.message.channel_mentions[0].id
+
 
 def setup(bot):
-    bot.add_cog(Mirror(bot))
+    bot.add_cog(Mirror(bot, bot.cache))
```

Comment:
it doesn't complain when load_extension is used?  i thought i had tried this and it gave me an argument error

Reply:
wait why do you need to pass it here if we already have it on `bot`?

Reply:
Seconded, this should just be derived from the bot

Reply:
moved it back into class